AzureMonitorAgent/agent.py

```diff
@@ -389,6 +389,10 @@ def enable():
     """
     global AMAServiceStartCommand, AMAServiceStatusCommand
 
+    # start the metrics process if its not already running
+    if (protected_settings is None or len(protected_settings) == 0) or (public_settings is not None and "proxy" in public_settings and "mode" in public_settings.get("proxy") and public_settings.get("proxy").get("mode") == "application"):
```

Comment:
nit: public_settings["proxy"] and public_settings["proxy"]["mode"] might be better for readability